server/src/storage/datastore.ts

```diff
@@ -38,49 +72,90 @@ const extractAndAppendId = (res: Entity) => {
 
 /**
  * A Datastore wrapper that gets a particular entity with the specified Kind and id.
- * @param kind The Kind that is being queried
- * @param id The id of the Entity being queried
+ * If actor is not specified, gets using datastore.
+ * @param kind The Kind that is being retrieved
+ * @param id The id of the Entity being retrieved
+ * @param actor Datastore or Transaction that will carry out the retrieval
```

Comment:
What's the difference between passing in Datastore and passing in Transaction? Also can we have a more specific name than actor

Reply:
@ifhuang You pass in a transaction object if you want that particular transaction to carry out the operation. The same transaction object has to be passed in for all operations that needs to be atomic. 
The datastore object (the default when nothing is passed in) just carries out the datastore operation (not in any transaction).

Reply:
@ifhuang Okay, so basically I'll be moving the actor code into the implementation? So I have this line on the first line of each function?
```
const actor = transaction || datastore;
```